ios/FluentUI/SegmentedControl/SegmentedControl.swift

```diff
@@ -117,21 +165,20 @@ open class SegmentedControl: UIView, TokenizedControlInternal {
 
         super.init(frame: .zero)
 
-        backgroundView.layer.cornerRadius = Constants.pillButtonCornerRadius
-        pillContainerView.addSubview(backgroundView)
+        stackView.layer.cornerRadius = Constants.pillButtonCornerRadius
+        pillContainerView.addSubview(stackView)
         selectionView.backgroundColor = .black
         pillContainerView.addSubview(selectionView)
         pillMaskedContentContainerView.mask = selectionView
-        pillMaskedContentContainerView.isUserInteractionEnabled = false
         pillContainerView.addSubview(pillMaskedContentContainerView)
         addButtons(items: items)
-        // We need to add pillMaskedContentContainerView to the container view
-        // before the buttons in order to activate the label constraints, but
-        // we want pillMaskedContentContainerView to show above the buttons.
-        pillContainerView.bringSubviewToFront(pillMaskedContentContainerView)
         pillContainerView.addInteraction(UILargeContentViewerInteraction())
-        addSubview(pillContainerView)
+        scrollView.addSubview(pillContainerView)
+        addSubview(gradientMask)
```

Comment:
the mask and scrollview shouldn't be part of the UIView hierarchy until those public apis are on? the facdt you made it variable also makes me think i can switch and it will rerender even after initialization?

Reply:
Which public APIs are you referring to? And they're vars instead of lets so that they could  be lazy vars and reference self during their setup. Are you saying we should dynamically remove and add the scroll view and mask based on `isFixedWidth`?

Reply:
Though, looks like the scrollView could still be a let. But the gradient mask needs to get the colors from self, so that's how it wound up as a lazy var.